pontoon/messaging/views.py

```diff
@@ -196,22 +206,38 @@ def get_recipients(form):
         approved = approved.filter(approved_date__lte=review_to)
         rejected = rejected.filter(rejected_date__lte=review_to)
 
-    approved = approved.values("approved_user").annotate(count=Count("approved_user"))
-    rejected = rejected.values("rejected_user").annotate(count=Count("rejected_user"))
+    # For the Minimum count, no value is the same as 0
+    # For the Maximum count, distinguish between no value and 0
+    if review_minimum or review_maximum is not None:
+        approved = approved.values("approved_user").annotate(
+            count=Count("approved_user")
+        )
+        rejected = rejected.values("rejected_user").annotate(
+            count=Count("rejected_user")
+        )
 
     if review_minimum:
         approved = approved.filter(count__gte=review_minimum)
         rejected = rejected.filter(count__gte=review_minimum)
 
-    if review_maximum:
+    if review_maximum is not None:
         approved = approved.filter(count__lte=review_maximum)
         rejected = rejected.filter(count__lte=review_maximum)
 
-    recipients = recipients.filter(
-        pk__in=list(submitted.values_list("user", flat=True).distinct())
-        + list(approved.values_list("approved_user", flat=True).distinct())
-        + list(rejected.values_list("rejected_user", flat=True).distinct())
-    )
+    if (
+        translation_from
+        or translation_to
+        or translation_minimum
+        or translation_maximum is not None
+    ):
+        submission_filters = list(submitted.values_list("user", flat=True).distinct())
+        recipients = recipients.filter(pk__in=submission_filters)
+
+    if review_from or review_to or review_minimum or review_maximum is not None:
+        review_filters = list(
+            approved.values_list("approved_user", flat=True).distinct()
+        ) + list(rejected.values_list("rejected_user", flat=True).distinct())
+        recipients = recipients.filter(pk__in=review_filters)
```

Comment:
Why listify the `submission_filters` and `review_filters`? Leaving them as just queries should also work?

Reply:
Updated. We should test if this affects performance.